addons/mail/static/src/new/activity/activity.js

```diff
@@ -39,20 +39,17 @@ export class Activity extends Component {
     async edit() {
         const { id, res_model, res_id } = this.props.data;
         await this.activity.scheduleActivity(res_model, res_id, id);
-        if (this.env.chatter) {
-            this.env.chatter.reload();
-        }
+        this.props.onUpdate();
     }
 
     async unlink() {
         await this.orm.unlink("mail.activity", [this.props.data.id]);
-        if (this.env.chatter) {
-            this.env.chatter.reload();
-        }
+        this.props.onUpdate();
     }
 }
 
 Object.assign(Activity, {
-    props: ["data"],
+    props: ["data", "onUpdate?"],
```

Comment:
Existence of `onUpdate` is not checked before call, it should be required.

Reply:
there is a default prop, so it is guaranteed to be defined. i think the activity component makes sense with an optional onUpdate prop